### DISCUSSION TITLE
What is best way to capture query latency?

### DISCUSSION THREAD

**Discussion by chanzlee** [TanStack/query#2467] [Q&A]
I'm trying to log query latency for service health check but I cannot find the good place to start. 

I see `onSettled` from default query options but there's no callback like `onFired` to capture the start time.

Should I create a wrapper hook for the queries and capture the start time there?

**Comment by TkDodo** [comment]
I would move this to the queryFn:

```
useQuery('key', async () => {
  captureStartTime()
  const { data } = await axios.get(....)
  captureEndTime()
  return data
})
```

or even move it to a different layer, like axios interceptors.
